### DISCUSSION TITLE
Proposal to improve DT ratechange leaderboard readability

### DISCUSSION THREAD

**Discussion by leggo15** [ppy/osu#29370] [Ideas]
With DT rate change coming to ranked, i propse using colors to make rates clear at a glance. These would follow the star rating coloring as found on the webpage. A play sat with 1.1x would for example have a bluemod icon.

My suggestion would be:

1.1x - Light blue

1.2x - Teal

1.3x - Green

1.4x - Yellow

1.5x - Red

1.6x - Magenta

1.7x - Pink

1.8x - Indigo

1.9x - Purple

2x - Black

This would help make leaderboards easy to read at a glance, instead of having to propperly read the rates of every play. I also think this would be a good solution for the website, currently website dt leaderboards are almost unusable since you have to hover over every play to see the rates used.

![chrome_pL15HxPKaH](https://github.com/user-attachments/assets/03b85372-df61-4a30-9b62-aefc81450cfd)
(basic example of how it could look)

**Comment by LacticWhale** [comment]
Maybe adding color different way might be better, like adding boarder, or changing inside part, or changing rate part (x1.1) color. That way red color will remain as indicator of difficulty increase mods.

And if this done properly can look really cool on all customizable mods (if make sense).

**Comment by ghost** [comment]
Remember that DT is not the only customizable mod, adding something like this would mean they will have to add it to other mods that have some sort of customization, and that would be a disaster since mods already have a set of colors each representing what they do to a map (increase/decrease difficulty, auto, etc), although it may look cool, its just better to rework the web/client leaderboard to show the number tag for DT and other customizable mods when possible like in the beatmap select menu leaderboard

**Reply by leggo15** [reply]
I dont think you nessesarily need colors for other mod customiztions, from what i garther people care fall less about which spesifications settings used for most mods, DT and HT is the only two i think needs a change like this. 

The time spent getting a sense of what rates are used and which stand out should be practically zero in my opinion, it should be as easy as it is to separate modded scores from eachother in standard, i think this change helps in that regard. Some people could probably give the current system a glance and have a good sense of a lb, but i really dont think that would be most people, some people have dyslexia(like me), some have bad vision, some have issues just building pattern recognission for numbers. I just think a color system would work better. ofc the actual rates should still display when you hover over each icon.

**Reply by Tenexxt** [reply]
> Remember that DT is not the only customizable mod, adding something like this would mean they will have to add it to other mods that have some sort of customization

Not true, in Lazer DT and HT mods have a 'tail' that has the speed value on it, unlike other mods with customisation, so it must've been decided that only speed changing mods will have such info displayed 'on sight', and for the rest you have to hover on it. Tho the website should show what mods were modified and what weren't by showing something near the mod icon, maybe a gear or 3 dots

**Reply by ghost** [reply]
Now that i think about it, what i said didn't make sense, but still i feel like changing color feels unnecessary, theres gotta be another way without breaking the rule of red color = hard

**Reply by Tenexxt** [reply]
Since the DT and HT have a tail displaying the multiplier (in lazer), it just needs to be implemented to the website (unless an other design than from Lazer is requested). The web has outdated mod designs anyway, so this could be implemented with the rest of the changes

**Comment by mixalisnew** [comment]
i think it would be maybe a better idea to adjust the 1.x amount color and keep all the other part of the mod red as it is

**Comment by Tenexxt** [comment]
IMO the colour change would make the leaderboards less readable, since different types of mods have different colours and this could cause confusion.
Tho in the game client DT scores with different speeds have a tail that has the speed multiplier on it. This should be added to the web page too, but personally I think the changing colours would be an interesting concept, so I decided to make this small design:
![IMG_20240813_011019](https://github.com/user-attachments/assets/4aba6a7e-9be4-45a7-9682-4913557f7f76)

I doubt that the colour change would get applied, but mod design REALLY need an update on the web (old icons, different mod colours, No Lazer Mods????)

**Comment by Stoppedpuma** [comment]
I think changing the colour Hue looks extremely messy and has no consistency with the rest of the design. I think a better path for a design direction such as this would be to change the brightness of the "tail" instead. Overall I'm against the change proposal in this discussion as I think just adding the tail to the leaderboard display is enough and keeps consistency with the rest of the design.

![image](https://github.com/user-attachments/assets/ae6f4a8d-7908-4c5d-a703-74f6322a2bc3)

**Reply by leggo15** [reply]
I feel like this wouldnt be enough to be able to see the rates at a glance. Coloring just the tail as Tenexxt proposed would work well i think.

**Reply by Tenexxt** [reply]
i think we need someone from the dev team reply to this because in the end they have to agree on the idea first, and then we can figure out the design.

**Reply by Stoppedpuma** [reply]
> Coloring just the tail as Tenexxt proposed would work well i think.

The problem here is inconsistency. Changing colours in this way is probably not going to happen for that reason. Additionally in my eyes there's no need for change beyond adding the number to the tail.

**Reply by Tenexxt** [reply]
in my opinion adding the tail with the value would be enough, tho as i already said - someone from the dev team needs to see this for this to progress in anyway.
Mod redesigns on the website are needed, so adding that tail would probably be done at the same time as the rest of the mods, but first this has to be aproved.
Also, some other icons beside mods are also outdated.
From what i checked the icons in the web can be found [here](https://github.com/ppy/osu-web/tree/master/public/images/badges/mods): 
I could replace them myself and do a pull request, **if i knew where in the game files are stored the current mod icons** (I looked now and can't find them, but im also in a rush right now and don't have time to look further), and i don't want to rely on outside sources like skins made after argon and other soures at all, **since they could be inaccurate.**
Also worth mentioning that i basically can't use PHP and JS, so i basically incapable for making the tail for DT and HT and other changes requested -_-

**Reply by Stoppedpuma** [reply]
The icons are already mostly up to date minus a small detail here or there. Flyte has a few redesigns for mods so if anything is delaying an update on the osu!web side it's probably that.
